### DISCUSSION TITLE
Uploading Base64 images to supabase

### DISCUSSION THREAD

**Discussion by Electric-Dragon** [supabase/supabase#6838] [Questions] [answered]
Hello,
My website wants to reduce the amount of things users can do with the images...so for added security, we want to keep the image sources as a base64 strings instead of a supabase storage URL. Is this possible through supabase storage, do will I have to save the string in the db?

**Comment by GaryAustin1** [comment]
You can use an arraybuffer to store base64 in storage.

https://supabase.com/docs/reference/javascript/storage-from-upload#upload-file-using-arraybuffer-from-base64-file-data

https://www.npmjs.com/package/base64-arraybuffer

**Reply by Electric-Dragon** [reply]
Okay...and when I call the `SELECT` method, will it return a `base64` string?

**Comment by GaryAustin1** [comment, marked as answer]
You would use download from storage.  If you want to go that root, you’ll have to search for someone who as done it.  Those were the only pointers I found to get you going.

**Reply by Electric-Dragon** [reply]
Okay...thanks a lot. I'll give it a try tomorrow

**Comment by Antiokh** [comment]
> Okay...thanks a lot. I'll give it a try tomorrow

Any success?
